nupic/torch/modules/sparse_weights.py

```diff
@@ -35,7 +35,7 @@ def rezero_weights(m):
 
     :param m: SparseWeightsBase module
     """
-    if isinstance(m, SparseWeightsBase):
+    if isinstance(m, HasRezeroWeights):
         if m.training:
```

Comment:
The `training` attribute is not part of  `HasRezeroWeights`. This will fail if used without `nn.Module`. To be safe check for both
```
if isinstance(m, (HasRezeroWeights, nn.Module)):
```

Reply:
Interesting, I didn't know we had this `m.training` check. That seems kinda broken. If someone calls `rezero_weights`, it seems like we should trust them and do it, regardless of `m.training`.

Reply:
Agree. We could just remove the `training` check

Reply:
Good catch. I believe that check would still pass (it's an either-or statement). I changed it to `isinstance(m, HasRezeroWeights) and isinstance(m, nn.Module)`

Reply:
I'll update it.